airframe-http-recorder/src/main/scala/wvlet/airframe/http/recorder/HttpRequestMatcher.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package wvlet.airframe.http.recorder
+
+import java.util.Locale
+
+import com.twitter.finagle.http.{MediaType, Request}
+import wvlet.log.LogSupport
+
+/**
+  * Compute a hash key of the given HTTP request.
+  * This value will be used for DB indexes
+  */
+trait HttpRequestMatcher {
+  def computeHash(request: Request): Int
+}
+
+object HttpRequestMatcher {
+
+  // Http headers to ignore for request hashing purposes
+  def defaultExcludeHeaderPrefixes: Seq[String] = Seq(
+    "date",           // unstable header
+    "x-b3-",          // Finagle's tracing IDs
+    "finagle-",       // Finagle specific headers
+    "host",           // The host value can be changed
+    "content-length", // this can be 0 (or missing)
+    "connection",     // Client might set this header
+    "user-agent"      // User-agent can be arbitrary
```

Comment:
To test authentication of multiple users and authentication failures, we should include this header to the hash value computation, but it will be excluded for recording for security.

Reply:
I see! It's not recorded but used for hash value computation. Makes sense.